### DISCUSSION TITLE
A better difftool for KiCad

### DISCUSSION THREAD

**Discussion by giscus** [cdwilson/cdwilson.dev#7] [Articles]
# A better difftool for KiCad

It can be difficult to tell what has actually changed when running git difftool on a .kicad_pcb file because KiCad embeds timestamps into the file that change when the file is modified:
Luckly, we can tell git to use a custom difftool using the --tool=<tool> argument.
First, we need to create a custom shell script to pass to git difftool --tool=<tool>. I use Kaleidoscope.app as my git difftool on macOS, so I named this script kicad-ksdiff and put it a directory on my $PATH:

https://cdwilson.dev/articles/a-better-difftool-for-kicad/

**Comment by danielsiegl** [comment]
Great article!

You can also run your script in textconv - I do this for PDF:

```
[diff "pdf"]
	textconv = sh -c 'date && pdftotext -layout -enc UTF-8 \"$0\" -'
	cachetextconv = true
```
